### PR TITLE
Delete children in release callbacks of Arrow CDataInterface

### PR DESCRIPTION
**Pull request by ankoh**
This PR extends the release callbacks of the CDataInterface to also cleanup memory of children.
Relevant Issue: https://github.com/duckdb/duckdb/issues/1612

The PR also includes a tiny commit that re-enables the archive APIs of miniz.
I want to expose miniz in the web version of DuckDB to extract arbitrary zips in the browser.
We'll bundle miniz anyway for the Parquet reader so we can provide fast wasm-driven zipping for free.
I can also move this to a separate PR if necessary.

### CODE REVIEW

File: src/common/types/data_chunk.cpp [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #include "duckdb/common/arrow.hpp"
 #include "duckdb/common/vector.hpp"
 #include "duckdb/common/to_string.hpp"
+#include <array>
```
Comment (Mytherin) [inline src/common/types/data_chunk.cpp:16]: Can we move this to "duckdb/common/array.hpp", similar to e.g. "duckdb/common/unordered_map.hpp", so we can avoid the std:: prefix and limit standard library includes to a single location?

**Comment by Mytherin** [comment]
Thanks! Looks good otherwise. I assume the miniz changes are related to #1591?

**Comment by ankoh** [comment]
Vaguely. #1591 is probably held back by the fact that miniz does not support streaming extraction at the moment.
https://github.com/duckdb/duckdb/blob/master/third_party/miniz/miniz.hpp#L1175

So we decided to do something more crude for wasm.
Instead of globbing into a zip as part of a scan, we're just providing a way to unzip and zip files manually for now.
But with that define there, we cant just use the bundled miniz of duckdb for that purpose.

#1591 demands for a glob like zip scan operator, I guess.

**Comment by hannesmuehleisen** [comment]
Thanks, looks great

**Pull request merged**